src/backend/distributed/utils/listutils.c

```diff
@@ -264,3 +264,20 @@ GenerateListFromElement(void *listElement, int listLength)
 
 	return list;
 }
+
+
+/*
+ * list_reverse returns a new list by reverting order of the elements within
+ * given list.
+ */
+List *
+list_reverse(const List *list)
+{
+	List *newList = NIL;
+	for (int i = list_length(list) - 1; i >= 0; i--)
+	{
+		newList = lappend(newList, list_nth(list, i));
```

Comment:
Calling `list_nth` for `n` times would imply `O(n^2)` runtime complexity in pg < 13 (due to underlying implementation), but I believe we will soon drop support for pg12 (probably in Citus-11 ?).

Reply:
[af529d3](https://github.com/citusdata/citus/pull/5692/commits/af529d3b5583851cf21f461fae3809ca2f7cbd3a)